### PR TITLE
test(advanced-color-picker): Playwright refactor

### PR DESCRIPTION
**Pull request by stephenogorman**
### Proposed behaviour

- Copy `advanced-color-picker` folder from `/cypress/locators/` to `/playwright/components/`
- Rename `index.js` to `index.ts` and `locators.js` to `locators.ts` in `/playwright/components/advanced-color-picker/` folder
- Refactor `index.ts` to Playwright
- Create `components.test-pw.tsx` in `/src/components/advanced-color-picker/` folder
- Move story components from `advanced-color-picker-test.stories.tsx' to `components.test-pw.tsx`
- Copy `advanced-color-picker.cy.tsx` from `/cypress/components/advanced-color-picker/` to `/src/components/advanced-color-picker/`
- Rename `advanced-color-picker.cy.tsx` to `advanced-color-picker.pw.tsx` in `/src/components/advanced-color-picker/` folder
- Refactor tests to Playwright

### Current behaviour

`Advanced Color Picker` tests are in Cypress not Playwright.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [x] Run Advanced Color Picker tests in Playwright Test Runner to check if the `advanced-color-picker.pw.tsx` file passed
- [x] Run all other tests in Playwright Test Runner to check none of the other `*.pw.tsx` files have regressed

### CODE REVIEW

File: src/components/advanced-color-picker/advanced-color-picker.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+        const picker = simpleColorPicker(page, 7);
+        await picker.press(key);
+        await expect(simpleColorPickerInput(page, index)).toHaveAttribute(
+          "aria-checked",
+          "true"
+        );
+      });
+    });
+
+    test(`should use key and move selection`, async ({ mount, page }) => {
```
Comment (ZhuoyuJin) [inline src/components/advanced-color-picker/advanced-color-picker.pw.tsx:79]: Shouldn't this test be included in the test line 56 - 77 above?

**Review by DipperTheDan — COMMENTED** [review]
Looks OK to me. I think you just need to delete the Cypress tests for this `Advanced Color Picker`.

File: src/components/advanced-color-picker/advanced-color-picker.pw.tsx (hunk 1 above)
Comment (stephenogorman) [reply src/components/advanced-color-picker/advanced-color-picker.pw.tsx:79]: Correct, that test was just a simplified version of the previous one added when I was trying to make it work. It should have been removed and is now.

**Review by ZhuoyuJin — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**